kernel/kernel-api/src/main/java/io/delta/kernel/ScanFile.java

```diff
@@ -0,0 +1,145 @@
+ * limitations under the License.
+ */
+package io.delta.kernel;
+
+import java.util.Map;
+
+import io.delta.kernel.client.TableClient;
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.fs.FileStatus;
+import io.delta.kernel.types.*;
+
+import io.delta.kernel.internal.actions.*;
+import io.delta.kernel.internal.fs.Path;
+
+/**
+ * Represents the metadata of {@link FilteredColumnarBatch} returned by
+ * {@link Scan#getScanFiles(TableClient)}. It contains and the schema of the columnar batch
+ * and methods to retrieve specific metadata values (e.x path in `AddFile` Delta Log action)
+ * from each scan file row in the batch.
+ */
+public class ScanFile {
+    /**
+     * Schema of the scan file row returned by {@link Scan#getScanFiles(TableClient)}.
+     * <p>
+     * <ol>
+     *  <li><ul>
+     *   <li>name: {@code add}, type: {@code struct}</li>
+     *   <li>Description: Represents `AddFile` DeltaLog action</li>
+     *   <li><ul>
+     *    <li>name: {@code path}, type: {@code string}, description: location of the file.</li>
+     *    <li>name: {@code partitionValues}, type: {@code map(string, string)},
+     *       description: A map from partition column to value for this logical file. </li>
+     *    <li>name: {@code size}, type: {@code log}, description: size of the file.</li>
+     *    <li>name: {@code modificationTime}, type: {@code log}, description: the time this
+     *       logical file was created, as milliseconds since the epoch.</li>
+     *    <li>name: {@code dataChange}, type: {@code boolean}, description: When false the
+     *      logical file must already be present in the table or the records in the added file
+     *      must be contained in one or more remove actions in the same version</li>
+     *    <li>name: {@code deletionVector}, type: {@code string}, description: Either null
+     *      (or absent in JSON) when no DV is associated with this data file, or a struct
+     *      (described below) that contains necessary information about the DV that is part of
+     *      this logical file. For description of each member variable in `deletionVector` @see
+     *      <a href=https://github.com/delta-io/delta/blob/master/PROTOCOL.md#Deletion-Vectors>
+     *          Protocol</a><ul>
+     *       <li>name: {@code storageType}, type: {@code string}</li>
+     *       <li>name: {@code pathOrInlineDv}, type: {@code string}</li>
+     *       <li>name: {@code offset}, type: {@code log}</li>
+     *       <li>name: {@code sizeInBytes}, type: {@code log}</li>
+     *       <li>name: {@code cardinality}, type: {@code log}</li>
+     *    </ul></li>
+     *   </ul></li>
+     *  </ul></li>
+     * </ol>
+     */
+    public static final StructType SCAN_FILE_SCHEMA = new StructType()
+        .add("add", AddFile.SCHEMA)
+        // TODO: table root is temporary, until the path in `add.path` is converted to
```

Comment:
Can we create a github issue with more details and point to it here?

Reply:
https://github.com/delta-io/delta/issues/2089